src/sort_transform.c

```diff
@@ -241,18 +189,23 @@ transform_int_op_const(OpExpr *op)
  * Note that if orig_expr(X) = orig_expr(Y) then
  *			 the ordering under new_expr is unconstrained.
  * */
-static Expr *
-sort_transform_expr(Expr *orig_expr)
+Expr *
+ts_sort_transform_expr(Expr *orig_expr)
 {
 	if (IsA(orig_expr, FuncExpr))
 	{
 		FuncExpr *func = (FuncExpr *) orig_expr;
 		char *func_name = get_func_name(func->funcid);
+		FuncInfo *finfo = ts_func_cache_get_bucketing_func(func->funcid);
+
+		if (NULL != finfo)
+		{
+			if (NULL == finfo->sort_transform)
+				return orig_expr;
+
+			return finfo->sort_transform(func);
+		}
 
-		if (strncmp(func_name, "date_trunc", NAMEDATALEN) == 0)
-			return transform_date_trunc(func);
-		if (strncmp(func_name, "time_bucket", NAMEDATALEN) == 0)
-			return transform_time_bucket(func);
 		if (strncmp(func_name, "timestamp", NAMEDATALEN) == 0)
```

Comment:
I guess these can now go into the cache? Also OK to leave for a separate PR.

Reply:
You mean the `timestamp` function? Figured it didn't have the right semantics to be considered a bucketing function, but if we'd like to turn this into a generic cache then I guess we can put it there. I'd rather do that separately though.